libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/routeprogress/RouteLegProgressTest.java

```diff
@@ -117,174 +77,129 @@ public void previousStep_returnsPreviousStep() {
-      .legIndex(0)
-      .build();
+  public void distanceTraveled_equalsZeroAtBeginning() throws Exception {
+    RouteProgress routeProgress = buildBeginningOfLegRouteProgress();
+
     assertEquals(0, routeProgress.currentLegProgress().distanceTraveled(), BaseTest.DELTA);
   }
 
   @Test
-  public void getDistanceTraveled_equalsLegDistanceAtEndOfLeg() {
-    RouteProgress routeProgress = RouteProgress.builder()
-      .stepDistanceRemaining(0)
-      .legDistanceRemaining(0)
-      .distanceRemaining(0)
-      .directionsRoute(route)
-      .stepIndex(firstLeg.steps().size() - 1)
-      .legIndex(0)
-      .build();
+  public void getDistanceTraveled_equalsLegDistanceAtEndOfLeg() throws Exception {
+    RouteProgress routeProgress = buildEndOfLegRouteProgress();
+    DirectionsRoute route = buildTestDirectionsRoute();
+    RouteLeg firstLeg = route.legs().get(0);
+
     assertEquals(firstLeg.distance(), routeProgress.currentLegProgress().distanceTraveled(), BaseTest.DELTA);
   }
 
   @Test
-  public void getDurationRemaining_equalsLegDurationAtBeginning() {
-    RouteProgress routeProgress = RouteProgress.builder()
-      .stepDistanceRemaining(route.legs().get(0).steps().get(0).distance())
-      .legDistanceRemaining(route.legs().get(0).distance())
-      .distanceRemaining(route.distance())
-      .directionsRoute(route)
-      .stepIndex(0)
-      .legIndex(0)
-      .build();
+  public void getDurationRemaining_equalsLegDurationAtBeginning() throws Exception {
+    RouteProgress routeProgress = buildBeginningOfLegRouteProgress();
+
     assertEquals(3535.2, routeProgress.currentLegProgress().durationRemaining(), BaseTest.DELTA);
   }
 
   @Test
-  public void getDurationRemaining_equalsZeroAtEndOfLeg() {
-    RouteProgress routeProgress = RouteProgress.builder()
-      .stepDistanceRemaining(0)
-      .legDistanceRemaining(0)
-      .distanceRemaining(0)
-      .directionsRoute(route)
-      .stepIndex(firstLeg.steps().size() - 1)
-      .legIndex(0)
-      .build();
+  public void getDurationRemaining_equalsZeroAtEndOfLeg() throws Exception {
+    RouteProgress routeProgress = buildEndOfLegRouteProgress();
+
     assertEquals(0, routeProgress.currentLegProgress().durationRemaining(), BaseTest.DELTA);
   }
 
   @Test
   public void followOnStep_doesReturnTwoStepsAheadOfCurrent() throws Exception {
-    RouteProgress routeProgress = RouteProgress.builder()
-      .stepDistanceRemaining(route.legs().get(0).steps().get(0).distance())
-      .legDistanceRemaining(route.legs().get(0).distance())
-      .distanceRemaining(route.distance())
-      .directionsRoute(route)
-      .stepIndex(5)
-      .legIndex(0)
-      .build();
+    RouteProgress routeProgress = buildDefaultTestRouteProgress();
+
+    routeProgress = routeProgress.toBuilder().stepIndex(5).build();
+
     assertTrue(routeProgress.currentLegProgress().followOnStep().geometry()
       .startsWith("un`ffAz_dogFz`Aq^hF{B|GaB|HcAxKKlIp@lOzC|Dh@hKtAzFh@`FDvHy@bG{AjKaEfF"));
```

Comment:
What's the value of testing geometries this way? Do geometries start always with that value? Could you clarify? If a test doesn't add any value :skull_and_crossbones: